ufo2fontir/src/source.rs

```diff
@@ -35,6 +26,59 @@ impl DesignSpaceIrSource {
     }
 }
 
+fn glif_files<'a>(
+    ufo_dir: &Path,
+    layer_cache: &'a mut HashMap<String, HashMap<String, PathBuf>>,
+    source: &designspace::Source,
+) -> Result<BTreeMap<String, PathBuf>, Error> {
+    let layer_name = layer_name(ufo_dir, layer_cache, source)?;
+    let glyph_dir = ufo_dir.join(layer_name);
+    if !glyph_dir.is_dir() {
+        return Err(Error::DirectoryExpected(glyph_dir));
+    }
+
+    let glyph_list_file = glyph_dir.join("contents.plist");
+    if !glyph_list_file.is_file() {
+        return Err(Error::FileExpected(glyph_list_file));
+    }
+    let result: BTreeMap<String, PathBuf> = plist::from_file(&glyph_list_file)
+        .map_err(|e| Error::ParseError(glyph_list_file, e.into()))?;
+
+    Ok(result
+        .into_iter()
+        .map(|(glyph_name, path)| (glyph_name, glyph_dir.join(path)))
+        .collect())
+}
+
+fn layer_contents(ufo_dir: &Path) -> Result<HashMap<String, PathBuf>, Error> {
+    let file = ufo_dir.join("layercontents.plist");
+    if !file.is_file() {
+        return Err(Error::FileExpected(file));
+    }
+    let contents: Vec<(String, PathBuf)> =
+        plist::from_file(&file).map_err(|e| Error::ParseError(file, e.into()))?;
+    Ok(contents.into_iter().collect())
+}
+
+pub(crate) fn layer_name<'a>(
+    ufo_dir: &Path,
+    layer_cache: &'a mut HashMap<String, HashMap<String, PathBuf>>,
+    source: &designspace::Source,
+) -> Result<&'a PathBuf, Error> {
+    static DEFAULT_LAYER_NAME: &str = "public.default";
+
+    if !layer_cache.contains_key(&source.filename) {
+        let contents = layer_contents(ufo_dir)?;
+        layer_cache.insert(source.filename.clone(), contents);
+    }
+    let layer_name = source.layer.as_deref().unwrap_or(DEFAULT_LAYER_NAME);
```

Comment:
this is incorrect. If source.layer is None, you should not assume that the default layer is called "public.default". The default layer is _default_ because the directory it maps to is called just "glyphs", but a default layer can be given any name, e.g. "foreground" or "rsheeter" if you like. Only when the default layer (the one associated with the "glyphs" directory) is not given a user-defined name, it defaults to "public.default".
See https://unifiedfontobject.org/versions/ufo3/layercontents.plist/#specification

You should do something similar to what I did in https://github.com/googlefonts/fontmake-rs/commit/2062c2f298907806be457716e63be013f8738bf8

```rust
        let mut layer_path = self.path.join("glyphs");
        if let Some(layer_name) = &self.layer {
            layer_path = self.path.join(
                self.layer_contents()?
                    .get(layer_name)
                    .ok_or_else(|| UfoToIrError::LayerNotFoundError(layer_name.to_string()))?,
            );
        }
```

Reply:
Ew. Ty, will fix.

Reply:
Fixed I think?